clue.js

```diff
@@ -0,0 +1,316 @@
+  image: './assets/conservatory.png'
+}
+
+const kitchen = {
+  name: 'Kitchen',
+  image: './assets/kitchen.png'
+}
+
+const study = {
+  name: 'Study',
+  image: './assets/study.png'
+}
+
+const library = {
+  name: 'Library',
+  image: './assets/library.png'
+}
+
+const billiardRoom = {
+  name: 'Billiard Room',
+  image: './assets/ball.png'
+}
+
+const lounge = {
+  name: 'Lounge',
+  image: './assets/lounge.png'
+}
+
+const ballroom = {
+  name: 'Ballroom',
+  image: './assets/ballroom.png'
+}
+
+const hall = {
+  name: 'Hall',
+  image: './assets/hall.png'
+}
+
+const spa = {
+  name: 'Spa',
+  image: './assets/spa.png'
+}
+
+const livingRoom = {
+  name: 'Living Room',
+  image: './assets/livingroom.jpg'
+}
+
+const observatory = {
+  name: 'Observatory',
+  image: './assets/observatory.png'
+}
+
+const theater = {
+  name: 'Theater',
+  image: './assets/theater.png'
+}
+
+const guestHouse = {
+  name: 'Guest House',
+  image: './assets/guesthouse.png'
+}
+
+const patio = {
+  name: 'Patio',
+  image: './assets/patio.png'
+}
+
+//GROUPING ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS
+
+const suspects = [
+  mrGreen,
+  mrsWhite,
+  professorPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard
+]
+
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+]
+
+const rooms = [
+  diningRoom,
+  conservatory,
+  kitchen,
+  study,
+  library,
+  billiardRoom,
+  lounge,
+  ballroom,
+  hall,
+  spa,
+  livingRoom,
+  observatory,
+  theater,
+  guestHouse,
+  patio
+]
+
+// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
+
+const randomSelector = array => {
+  return array[Math.floor(Math.random() * array.length)]
+}
+
+// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
+
+let mysteryobject = {
+  killer: "",
+  weapon: "",
+  room: "",
+}
+
+// This function will be invoked when you click on the killer card.
+const pickKiller = () => {
+  // This will randomly select a killer from the suspects. And add that to the mystery object.
+  mystery.killer = randomSelector(suspects)
+
+  // This will change the background color of the card to the one connected to the chosen killer and show selected elements
+  document.getElementById('killerCard').style.background = mystery.killer.color
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerImage').src = `${mystery.killer.image}`
+  document.getElementById('killerAge').innerHTML = `Age: ${mystery.killer.age}`
+  document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`
+  document.getElementById('killerDescription').innerHTML = `Description: ${mystery.killer.description}`
+  document.getElementById("killerFavouriteWeapon").innerHTML = `Favourite weapon: ${shuffleFavouriteWeapon()}`
+}
+
+document.getElementById('killerCard').addEventListener('click', pickKiller)
```

Comment:
This could also have been solved by adding the onclick attribute in the HTML-element:
` <article class="killer-deck" id="killerCard" onclick="pickKiller()">`

Like you have done on your reveal-button!
`<button onclick="revealMystery()">REVEAL THE CRIME</button>`

Reply:
It could, but don't have to - both options are ok